x/gov/keeper/deposit.go

```diff
@@ -162,6 +163,73 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// chargeDeposit will charge proposal cancellation fee (deposits * proposal_cancel_burn_rate)  and
+// send to a destAddress if defined or burn otherwise.
+// Remaining funds are send back to the depositor.
+func (keeper Keeper) chargeDeposit(ctx sdk.Context, proposalID uint64, destAddress, proposalCancelRate string, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	rate := sdk.MustNewDecFromStr(proposalCancelRate)
+	var cancellationCharges sdk.Coins
+
+	if rate.IsPositive() {
```

Comment:
I think this will return `true` for `zero` value too, @gsk967 can you recheck?

Reply:
Yes, It will return `true` when value must be greater than 0

Reply:
@atheeshp , `IsPositive()` returns false for zero value.